From f46eb8d3f8183ea223b03774ae48fed8b2523339 Mon Sep 17 00:00:00 2001
From: Hu Wang <quic_huw@quicinc.com>
Date: Thu, 19 Sep 2024 18:58:18 -0700
Subject: [PATCH] SAE: Extend Basic MLE Recognition to external auth case

This commit extends ccba6921de63 ("SAE: Recognize Basic MLE in
Authentication frames even without H2E") to cover external auth case.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 wpa_supplicant/sme.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/sme.c b/wpa_supplicant/sme.c
index bf0252b91..05ca8ddad 100644
--- a/wpa_supplicant/sme.c
+++ b/wpa_supplicant/sme.c
@@ -1726,7 +1726,9 @@ static int sme_sae_auth(struct wpa_supplicant *wpa_s, u16 auth_transaction,
 			}
 			token_len = elen - 1;
 #ifdef CONFIG_IEEE80211BE
-		} else if (wpa_s->valid_links && token_len > 12 &&
+		} else if ((wpa_s->valid_links ||
+			    (external && wpa_s->sme.ext_ml_auth)) &&
+			   token_len > 12 &&
 			   token_pos[token_len - 12] == WLAN_EID_EXTENSION &&
 			   token_pos[token_len - 11] == 10 &&
 			   token_pos[token_len - 10] ==
-- 
2.43.0

